### DISCUSSION TITLE
Sidebar Link does not show image from classname

### DISCUSSION THREAD

**Discussion by shipswake** [facebook/docusaurus#10638] [General] [answered]
I have a class that displays a coming soon image to a doc in the sidebar:

```
{
    type: 'doc',
    id: 'explorers',
    label: 'Explorer',
    className: 'coming-soon'
 },
```

```
.coming-soon > div > a::after {
  content: url(../../static/img/coming-soon.svg);
  border-top: solid 5px transparent;
  padding-left: 0.3rem;
}
```

When I apply this class name to a link I do not see the image
```
[{
    type: 'category',
    label: 'Order Management',
    link: { type: 'doc', id: 'order-management-intro' },
    className: 'coming-soon',
    collapsed: false,
    items: []
 }]
 ```

 Is there a way to accomplish this?

**Comment by shipswake** [comment]
I see in the [docs](https://docusaurus.io/docs/sidebar#complex-sidebars-example) there is an image prop on the link sidebar item type but it does not seem to be on the [link](https://docusaurus.io/docs/sidebar/items#sidebar-item-link
)

**Comment by slorber** [comment]
maybe related to CSS nesting bug: https://github.com/facebook/docusaurus/issues/9303

**Reply by shipswake** [reply]
I am assuming there is a fix in the 3.6 version or do I need to do something to workaround?

**Reply by slorber** [reply]
It's not really a fix in v3.6, but I think turning on faster / LightningCSS might fix the issue as a side effect. 

The bug remains here when not using Lightning CSS, so it's still a bug we have to solve for those can can't turn these new options on.

**Reply by shipswake** [reply]
I'll try turning on Lightening CSS

**Reply by shipswake** [reply]
Using 3.6 with experimental_faster: true and I still do not see the className displayed

**Reply by slorber** [reply, marked as answer]
So, maybe I was wrong, or your code is wrong

**Reply by shipswake** [reply]
The issue seems to be the hierarchy between the type: doc and the type: category
category contains a div I was not accounting for

**Reply by shipswake** [reply]
with the above code I am able to get the image to show correctly

![image](https://github.com/user-attachments/assets/0ce6cb95-7aa6-4a91-aba7-60d7967aba94)


When I expand the category the image rotates because of this css

![image](https://github.com/user-attachments/assets/87ee5104-9ed8-4325-a77e-0d8dda28135f)

```
.menu__list-item--collapsed .menu__caret:before {
    transform: rotateZ(90deg);
}
```

there seems to be a conflict with the a::after and caret::before css
```
.coming-soon > div > a::after {
  content: url(../../static/img/coming-soon.svg);
  border-top: solid 5px transparent;
  padding-left: 0.3rem;
}
```

Is there a way to stop the image from rotating when the caret rotates?

**Reply by slorber** [reply]
For more advanced customizations, it may be better to swizzle and alter the HTML structure rather than trying to piggy pack on our existing HTML structure with complex CSS selectors. 

I don't know exactly what you should do, but keep in mind those things are implementation details, and we might change the underlying HTML markup or CSS at any time.

**Reply by shipswake** [reply]
Trying to swizzle DockSidebarItem\Category but it does not seem to pick it up?
Is this the right object to swizzle?

**Reply by slorber** [reply]
Yes it should be the right one. You might need to restart the dev server after swizzle, we have a little hot reload issue when new theme files are created.

**Reply by shipswake** [reply]
I made a copy in the theme folder and that did not work

I run the command `docusaurus swizzle @docusaurus/theme-classic/DocSidebarItem`

and I get an error

```
[ERROR] Theme @docusaurus/theme-classic/DocSidebarItem not found. Themes available for swizzle: 
- @docusaurus/theme-classic
- @docusaurus/plugin-debug
- @easyops-cn/docusaurus-search-local
```

I've done this before with the Navbar and Footer so I'm not sure what is happening

**Reply by slorber** [reply]
Because this is now how to use the Cli. Just run docusaurus swizzle and let it guide you.
